lib/streamlit/caching.py

```diff
@@ -53,6 +54,18 @@
 to suppress the warning.
 """
 
+# based on: https://stackoverflow.com/questions/43506378/how-to-get-source-code-of-function-that-is-wrapped-by-a-decorator
+# In Python 2, the @functools.wraps() decorator does not set the convenience __wrapped__
+if sys.version_info[0:2] >= (3, 4):  # Python v3.4+?
+    wraps = functools.wraps  # built-in has __wrapped__ attribute
+else:
+    def wraps(wrapped, assigned=functools.WRAPPER_ASSIGNMENTS,
```

Comment:
Should we add some tests for python 2?

Reply:
Well, if this doesn't work, a caching test will fail.